### PR TITLE
fix new_thread join

### LINKED ISSUE #645: Connect example output different than expected
Hi, 

Thanks for the great repo!

I tried out your [connect ](https://github.com/victimsnino/ReactivePlusPlus/blob/v2/src/examples/rpp/doxygen/connect.cpp)example as is on [godbolt](https://godbolt.org/z/a6qcsP7KG) but the output seems different. It never reaches printing the CONNECT message. Is that expected?

Regards,
Willem

### PR DESCRIPTION
**Pull request by victimsnino**
Fix #645

## Summary by CodeRabbit

- **New Features**
	- Simplified thread management by ensuring threads are always detached.

- **Bug Fixes**
	- Enhanced test case structure in the scheduler tests for better clarity and synchronization.

- **Refactor**
	- Updated the `wait` function in mock observer to accept a more generic template for expectations.

### CODE REVIEW

File: src/rpp/rpp/schedulers/new_thread.hpp [hunk 1]
```diff
@@ -44,11 +44,7 @@ namespace rpp::schedulers
                     m_state->is_stoping = true;
                 }
                 m_state->cv.notify_all();
-
-                if (m_thread.get_id() != std::this_thread::get_id())
-                    m_thread.join();
-                else
-                    m_thread.detach();
+                m_thread.detach();
```
[the remark this answers is not in the document]
Comment (victimsnino) [reply src/rpp/rpp/schedulers/new_thread.hpp:47]: It is expected behavior, we just mark in destructor that no anyone is going to submit new task, but it is OK to finish processing of tasks from queue if any

**Review by wjspoel — APPROVED** [review]
